In [1]:
import pandas as pd
import json
from pandas import json_normalize
import requests
import time

Get all comments between January 1, 2019 at 00:00:00 (1546300800)
and December 31, 2020 at 23:59:59 (1577836799)
`
Mini test: until January 7, 2019 at 23:59:59 (1546905599)

In [2]:
df_list = []
iteration_count = 0

In [3]:
current_time = 1546300800 # Jan 1st, 2019
end_time =  1562025600 # June 2nd, 2019
# subreddit = "wallstreetbets"
# size = 1000

link_holder = "https://api.pushshift.io/reddit/search/comment/?subreddit=wallstreetbets&after={}&before={}&size=1000"

In [4]:
t0 = time.time()
while current_time < end_time:
    r = requests.get(link_holder.format(current_time, end_time))
        
    if r.json()['data'] == []:
        print('empty')
        break
    df = json_normalize(r.json(), "data")
    df_list.append(df)
    
    # get the newest comment in the data frame and set it to the current time 
    # so next time we get a new set of 500 comments
    
    time.sleep(.4)
    current_time = df["created_utc"].max()
    if iteration_count % 20 == 0:
        print(iteration_count, current_time)
    else:
        print(iteration_count, current_time, end=" ")
    iteration_count += 1
t1 = time.time()

0 1546322457
1 1546362502 2 1546383218 3 1546397988 4 1546409687 5 1546436345 6 1546443079 7 1546447997 8 1546455233 9 1546460306 10 1546465345 11 1546468573 12 1546474078 13 1546480870 14 1546489533 15 1546498397 16 1546518245 17 1546526731 18 1546530348 19 1546534288 20 1546538838
21 1546543548 22 1546547545 23 1546551207 24 1546557924 25 1546566400 26 1546575148 27 1546592499 28 1546611407 29 1546615472 30 1546618558 31 1546622636 32 1546626869 33 1546631124 34 1546635453 35 1546642839 36 1546655603 37 1546677628 38 1546711911 39 1546731369 40 1546754521
41 1546793413 42 1546810768 43 1546823666 44 1546835246 45 1546867339 46 1546875560 47 1546881004 48 1546886861 49 1546892894 50 1546901084 51 1546911037 52 1546923018 53 1546952191 54 1546959445 55 1546963755 56 1546969235 57 1546975482 58 1546981100 59 1546988020 60 1546997336
61 1547002204 62 1547010523 63 1547032260 64 1547044993 65 1547049218 66 1547053738 67 1547057259 68 1547061262 69 1547065902 70 1547069669 71 1547077605 72

541 1552876760 542 1552911733 543 1552920784 544 1552927914 545 1552936950 546 1552945427 547 1552957380 548 1552969637 549 1552998709 550 1553005053 551 1553010628 552 1553015169 553 1553021151 554 1553025797 555 1553033008 556 1553044036 557 1553055748 558 1553084556 559 1553091472 560 1553096776
561 1553101369 562 1553105510 563 1553109056 564 1553113278 565 1553120029 566 1553129871 567 1553143631 568 1553170616 569 1553176744 570 1553180864 571 1553185545 572 1553190150 573 1553194195 574 1553198215 575 1553204078 576 1553211771 577 1553221272 578 1553236135 579 1553259799 580 1553264365
581 1553268886 582 1553272406 583 1553276271 584 1553281022 585 1553285660 586 1553292970 587 1553306828 588 1553345382 589 1553362971 590 1553382485 591 1553414175 592 1553449751 593 1553464848 594 1553478706 595 1553492031 596 1553515656 597 1553522739 598 1553529006 599 1553534484 600 1553537846
601 1553541844 602 1553548481 603 1553556536 604 1553565746 605 1553578170 606 1553604910 607 155361

1081 1557466082 1082 1557487163 1083 1557492877 1084 1557496731 1085 1557499861 1086 1557502786 1087 1557505743 1088 1557509058 1089 1557512163 1090 1557515913 1091 1557518889 1092 1557524690 1093 1557534573 1094 1557548991 1095 1557584134 1096 1557600096 1097 1557612340 1098 1557623883 1099 1557653073 1100 1557685165
1101 1557699492 1102 1557709529 1103 1557722255 1104 1557748381 1105 1557753494 1106 1557756779 1107 1557760644 1108 1557764294 1109 1557767643 1110 1557770955 1111 1557775472 1112 1557779181 1113 1557784561 1114 1557791570 1115 1557799317 1116 1557809329 1117 1557832821 1118 1557840805 1119 1557846023 1120 1557851375
1121 1557857135 1122 1557863161 1123 1557867349 1124 1557874794 1125 1557884865 1126 1557894633 1127 1557919249 1128 1557927046 1129 1557931198 1130 1557935606 1131 1557940950 1132 1557945944 1133 1557950772 1134 1557956101 1135 1557963354 1136 1557972328 1137 1557985144 1138 1558009371 1139 1558015480 1140 1558019511
1141 1558024068 1142 1558028578 1143 155

In [5]:
t1 = time.time()
print('Took %f seconds to run' % (t1-t0))

Took 2680.284825 seconds to run


Takes 45 minutes to run

In [6]:
comments = pd.concat(df_list, ignore_index=True)
comments

author  author_created_utc author_flair_background_color  \
0                    Nm0369        1.493507e+09                          None   
1               achennupati        1.487615e+09                          None   
2                 grissomza        1.443582e+09                          None   
3                ktdfintech        1.528505e+09                          None   
4          Lester_Diamond23        1.503953e+09                          None   
...                     ...                 ...                           ...   
1534377  OrangesAreOrangeHa                 NaN                          None   
1534378       EctasyFantasy                 NaN                          None   
1534379        PowerfulLier                 NaN                          None   
1534380         RadiationDM                 NaN                          None   
1534381            CPA-hole                 NaN                          None   

        author_flair_css_class author_flair_richtext author_flair_template_id  \
0                         None                    []                     None   
1                         None                    []                     None   
2                         None                    []                     None   
3                         None                    []                     None   
4                         None                    []                     None   
...                        ...                   ...                      ...   
1534377                   None                    []                     None   
1534378                   None                    []                     None   
1534379                   None                    []                     None   
1534380                   None                    []                     None   
1534381                   None                    []                     None   

        author_flair_text author_flair_text_color author_flair_type  \
0                    None                    None              text   
1                    None                    None              text   
2                    None                    None              text   
3                    None                    None              text   
4                    None                    None              text   
...                   ...                     ...               ...   
1534377              None                    None              text   
1534378              None                    None              text   
1534379              None                    None              text   
1534380              None                    None              text   
1534381              None                    None              text   

        author_fullname  ... gildings.gid_2 gildings.gid_3  author_cakeday  \
0             t2_17iqgr  ...            0.0            0.0             NaN   
1             t2_15l23j  ...            0.0            0.0             NaN   
2              t2_qtsq4  ...            0.0            0.0             NaN   
3           t2_1jas53nt  ...            0.0            0.0             NaN   
4            t2_c14dcw5  ...            0.0            0.0             NaN   
...                 ...  ...            ...            ...             ...   
1534377     t2_1oebvjgx  ...            NaN            NaN             NaN   
1534378     t2_3jbd43r3  ...            NaN            NaN             NaN   
1534379        t2_f79kf  ...            NaN            NaN             NaN   
1534380        t2_bfm99  ...            NaN            NaN             NaN   
1534381       t2_1588uf  ...            NaN            NaN             NaN   

        distinguished edited  gilded is_submitter locked  all_awardings  \
0                 NaN    NaN     NaN          NaN    NaN            NaN   
1                 NaN    NaN     NaN          NaN    NaN            NaN   
2                 NaN    NaN     NaN          NaN    NaN    

In [7]:
len(comments)

1534382

In [8]:
current_time

1562025593

In [9]:
# the first 500 comments were spread over a time of 9050 seconds 
comments['created_utc'].max() 

1562025593

In [10]:
display(comments['created_utc'].sort_values()[1000] - 1546300800)

(end_time - 1546300800) / iteration_count

21662

10244.169381107491

In [13]:
comments.to_csv('Jan1-Jun2_2019_comments.csv')

In [12]:
comments.to_csv('Jan1-Jun2_2019_comments_no_index.csv')

The first request gave us 21662 seconds worth of comments (360 minute)

On average each request gave us 10244 seconds worth of comments comments (170 minutes)